proxy/src/timeout.rs

```diff
@@ -92,7 +89,7 @@ where
         let duration = self.duration;
         // TODO: should this panic or wrap the error?
         let timeout = ReactorTimeout::new(duration, &self.handle)
-            .expect("failed to create timeout!");
+            .expect("reactor gone");
```

Comment:
I guess with the expect string explaining why an error is unexpected, the `TODO` comment above could go away?